### DISCUSSION TITLE
[useForm] A Few suggestions for improvement

### DISCUSSION THREAD

**Discussion by josch0** [mantinedev/mantine#2090] [Feature requests]
I use useForm and like the simplicity. It provides everything that a simple form handling needs.

There are a few small things that I noticed and would like to suggest as improvements:

1.) Add "hasErrors (true/false)" to the useForm response. Since "errors" is always a valid object, it is cumbersome to find out if it actually contains keys/errors.

2.) Adding an "onChange" listener to the useForm response. When I change the values of some inputs, I need to reset the values in other dependent inputs. At the moment I don't know how to solve this easily.

3.) Add the ability to add the "is required" asterisk inside the label without the HTMLComponent actually setting "required". I don't want the browser to check this, as it prevents the useForm validation and thus I don't get any errors in the validation response. 

Many thanks in advance!
Mantine is awesome! :-)

**Comment by rtivital** [comment]
1) Will be added in 5.2
2) Not sure about it, I'm thinking about `useFieldEffect` hook, need to test it with strict mode
3) Will be added in 5.x – all inputs will have `withRequiredAsterisk` prop

**Comment by josch0** [comment]
In my opinion, onChange would fit well with onReset and onSubmit. This would allow you to add your own logic at any point in the editing process. So why have a separate hook for it if you are using useForm anyway?

To make the example a bit more concrete. I need to clear a TextInput when the user unchecks a checkbox.
What would be your suggestion to implement this?

```
form.onChange(() => {
  if (form.values.fooCheckbox === false) {
      form.setValue('barTextInput', '')
  }
})
```

**Reply by rtivital** [reply]
Most likely it is not possible to make it work this way. Events in react are handled in useEffect. Your code snippet is the same as `useEffect(() => {}, [form.values]);`
